File: backend/src/main/java/com/project/dto/auth/request/RegisterProfessionalDTO.java
package com.project.dto.auth.request;

import java.math.BigDecimal;

import com.project.enums.Gender;
import com.project.enums.Specialization;
import com.project.enums.SpokenLanguage;

import jakarta.validation.constraints.DecimalMin;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.Max;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Pattern;
import jakarta.validation.constraints.Size;

import lombok.Data;

@Data
public class RegisterProfessionalDTO {

    @NotBlank(message = "Full name is required")
    @Size(min = 3, max = 50, message = "Full name must be between 3 and 50 characters")
    @Pattern(
        regexp = "^[A-Za-z ]+$",
        message = "Full name can contain only letters and spaces"
    )
    private String fullName;

    @NotBlank(message = "Email is required")
    @Email(message = "Invalid email format")
    private String email;

    @NotBlank(message = "Password is required")
    @Size(min = 6, max = 64, message = "Password must be between 6 and 64 characters")
    @Pattern(
        regexp = "^(?=.*[a-z])(?=.*[A-Z])(?=.*\\d)(?=.*[@$!%*?&])[A-Za-z\\d@$!%*?&]+$",
        message = "Password must contain at least 1 uppercase, 1 lowercase, 1 number, and 1 special character"
    )
    private String password;

    @NotBlank(message = "Phone number is required")
    @Pattern(
        regexp = "^[6-9]\\d{9}$",
        message = "Phone number must be a valid 10-digit Indian mobile number"
    )
    private String phone;

    @NotNull(message = "Gender is required")
    private Gender gender;

    @Pattern(
        regexp = "ROLE_PROFESSIONAL",
        message = "Role must be ROLE_PROFESSIONAL"
    )
    private String role = "ROLE_PROFESSIONAL";

    @NotNull(message = "Specialization is required")
    private Specialization specialization;

    @Min(value = 0, message = "Experience cannot be negative")
    @Max(value = 60, message = "Experience seems invalid")
    private int experienceYears;

    @NotBlank(message = "Qualification is required")
    @Size(min = 2, max = 100, message = "Qualification must be between 2 and 100 characters")
    private String qualification;

    @Size(max = 500, message = "Bio must not exceed 500 characters")
    private String bio;

    @NotNull(message = "Consultation fee is required")
    @DecimalMin(value = "0.0", inclusive = false, message = "Consultation fee must be greater than 0")
    private BigDecimal consultationFee;

    @NotNull(message = "Spoken language is required")
    private SpokenLanguage spokenLanguage;
}

File: backend_dotnet/SafeMind.Core/DTOs/Auth/RegisterProfessionalDTO.cs
using System.ComponentModel.DataAnnotations;
using SafeMind.Core.Enums;

namespace SafeMind.Core.DTOs.Auth
{
    public class RegisterProfessionalDTO
    {
        [Required(ErrorMessage = "Full name is required")]
        [StringLength(50, MinimumLength = 3, ErrorMessage = "Full name must be between 3 and 50 characters")]
        [RegularExpression(@"^[A-Za-z .'-]+$", ErrorMessage = "Full name can contain only letters, spaces, dots, hyphens and apostrophes")]
        public string FullName { get; set; } = string.Empty;

        [Required(ErrorMessage = "Email is required")]
        [EmailAddress(ErrorMessage = "Invalid email format")]
        public string Email { get; set; } = string.Empty;

        [Required(ErrorMessage = "Password is required")]
        [StringLength(64, MinimumLength = 6, ErrorMessage = "Password must be between 6 and 64 characters")]
        [RegularExpression(@"^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[@$!%*?&.#^()_+=<>-]).{6,64}$", ErrorMessage = "Password must contain at least 1 uppercase, 1 lowercase, 1 number, and 1 special character")]
        public string Password { get; set; } = string.Empty;

        [Required(ErrorMessage = "Phone number is required")]
        [RegularExpression(@"^[6-9]\d{9}$", ErrorMessage = "Phone number must be a valid 10-digit Indian mobile number")]
        public string Phone { get; set; } = string.Empty;

        [Required(ErrorMessage = "Gender is required")]
        public Gender Gender { get; set; }

        public string Role { get; set; } = "ROLE_PROFESSIONAL";

        [Required(ErrorMessage = "Specialization is required")]
        public Specialization Specialization { get; set; }

        [Range(0, 60, ErrorMessage = "Experience seems invalid")]
        public int ExperienceYears { get; set; }

        [Required(ErrorMessage = "Qualification is required")]
        [StringLength(100, MinimumLength = 2, ErrorMessage = "Qualification must be between 2 and 100 characters")]
        public string Qualification { get; set; } = string.Empty;

        [StringLength(500, ErrorMessage = "Bio must not exceed 500 characters")]
        public string? Bio { get; set; }

        [Required(ErrorMessage = "Consultation fee is required")]
        [Range(0.1, double.MaxValue, ErrorMessage = "Consultation fee must be greater than 0")]
        public decimal ConsultationFee { get; set; }

        [Required(ErrorMessage = "Spoken language is required")]
        public SpokenLanguage SpokenLanguage { get; set; }
    }
}
